instrumentation/jdbc/library/src/test/java/io/opentelemetry/instrumentation/jdbc/datasource/JdbcTelemetryTest.java

```diff
@@ -66,6 +74,48 @@ void buildWithDefaults() throws SQLException {
         SERVER_PORT);
   }
 
+  @Test
+  void error() throws SQLException {
+    assumeTrue(SemconvStability.emitStableDatabaseSemconv());
+
+    JdbcTelemetry telemetry = JdbcTelemetry.builder(testing.getOpenTelemetry()).build();
+    DataSource source = spy(new TestDataSource());
+    Connection connection = spy(source.getConnection());
+    Statement statement = spy(connection.createStatement());
+    when(source.getConnection()).thenReturn(connection);
+    when(connection.createStatement()).thenReturn(statement);
+    doThrow(new SQLException("BOOM", "state", 42)).when(statement).execute(Mockito.anyString());
+    DataSource dataSource = telemetry.wrap(source);
+
+    assertThatCode(
+            () ->
+                testing.runWithSpan(
+                    "parent",
+                    () -> dataSource.getConnection().createStatement().execute("SELECT 1;")))
+        .isInstanceOf(SQLException.class);
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("parent"),
+                span -> span.hasName("TestDataSource.getConnection"),
+                span ->
+                    span.hasName("SELECT dbname")
+                        .hasAttribute(equalTo(DB_RESPONSE_STATUS_CODE, "42"))
+                        .hasAttribute(equalTo(ERROR_TYPE, "java.sql.SQLException"))));
```

Comment:
we could be a little more explicit in this assertion:

```suggestion
                    span.hasName("SELECT dbname")
                        .hasAttributesSatisfyingExactly(
                            equalTo(DB_SYSTEM_NAME, "postgresql"),
                            equalTo(DB_OPERATION_NAME, "SELECT"),
                            equalTo(DB_NAMESPACE, "dbname"),
                            equalTo(DB_QUERY_TEXT, "SELECT ?;"),
                            equalTo(DB_RESPONSE_STATUS_CODE, "42"),
                            equalTo(SERVER_ADDRESS, "127.0.0.1"),
                            equalTo(SERVER_PORT, 5432),
                            equalTo(ERROR_TYPE, "java.sql.SQLException"))));
```